src/aws/builder.rs

```diff
@@ -891,6 +893,12 @@ impl AmazonS3Builder {
         self
     }
 
+    /// Set the max keys per list request. It's almost used for test paginated listing.
```

Comment:
This functionality never appears to be used or tested, is it necessary?

Reply:
yeah, i'm trying to test paginated listing, for example, limit n objects per list request instead of preparing more than 1k objects before testing in ut, but i found it's a little bit hard to add a general test case in `integration.rs` because they used a common object store, e.g. AmazonS3, otherwise I need to create separate store and test cover it. It can be removed.